csrc/ops/composite.cpp

```diff
@@ -276,13 +276,13 @@ static TensorView* newForMatmul(TensorView* tv_a, TensorView* tv_b) {
   auto ndims_b = orig_domain_b.size();
 
   // Matmul output size is same as the higher dimensional input size if both A/B
-  // > 1D.
-  auto ndims_out = std::max(ndims_a, ndims_b);
+  // > 1D, but with 1 additional IterType::Reduction axis rK.
+  auto ndims_out = std::max(ndims_a, ndims_b) + 1;
```

Comment:
Changes in this file are for adding a new `Reduction` output domain to facilitate easier mapping of K axes.